File: impl-go/no_0070_climbing_stairs/no_0070_climbing_stairs.go
package no_0070_climbing_stairs

func climbStairs(n int) int {
	// 就一级阶梯，所以只有一种方法
	if n == 1 {
		return 1
	}

	first := 1  // (0, 1)
	second := 2 // (0, 2)
	for i := 3; i <= n; i++ {
		sum := first + second
		first = second
		second = sum
	}
	return second
}

File: impl-rust/src/no_0070_climbing_stairs.rs

pub fn climb_stairs(n: i32) -> i32 {
    if n == 1 {
        return 1;
    }

    let mut first = 1;
    let mut second = 2;
    for _ in 3..=n {
        let sum = first+second;
        first = second;
        second = sum;
    }
    second
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_climb_stairs() {
        assert_eq!(1, climb_stairs(1));
        assert_eq!(2, climb_stairs(2));
        assert_eq!(3, climb_stairs(3));
        assert_eq!(5, climb_stairs(4));
    }
}